wurst/_handles/DialogBox.wurst

```diff
@@ -0,0 +1,67 @@
+package DialogBox
+import HashMap
+import Dialog
+import LinkedList
+
+/**DialogBox lets you to create dialogs and associate the buttons with code.*/
+public class DialogBox
+
+    private static constant HashMap<button, trigger> buttonTriggers = new HashMap<button, trigger>
+    private static constant HashMap<dialog, DialogBox> dialogBoxes = new HashMap<dialog, DialogBox>
+    private constant LinkedList<button> buttons = new LinkedList<button>
+
+    private constant dialog dialogHandle = createDialog()
+    private constant trigger clickDialog = CreateTrigger()
+        ..registerDialogEvent(this.dialogHandle)
+        ..addAction(function removeClickedDialog)
+
+    private string title
+    
+
+    construct(string title)
+        this.title = title
+        this.dialogHandle.setMessage(title)
+        dialogBoxes.put(this.dialogHandle, this)
+
+    private static function removeClickedDialog()
+        let btn = GetClickedButton()
+        buttonTriggers.get(btn).execute()
+        destroy dialogBoxes.get(GetClickedDialog())
+
+    function addButton(string buttonText, code cb)
+        this.addButton(buttonText, 0 , cb)
+
+    /**Hotkey: use ASCII number of the capital letter.*/
+    function addButton(string buttonText, integer hotkey, code cb)
```

Comment:
The general consensus here should be to use a closure instead of code, so the user does not have to worry about carrying over data.
Imagine following freehand example (didnt test):
```
enum UpgType
    A
    B
    C
class UpgradeShop
    DialogBox dialogBox = new DialogBox("Select Upgrade")
    unit buyingUnit = null

    construct()
        dialogBox..addButton("A", () -> onUpgrade(UpgType.A))
        ..addButton("B", () -> onUpgrade(UpgType.B))
        ..addButton("C", () -> onUpgrade(UpgType.C))
        
    function onUpgrade(UpgType utype)
        dialogBox..display(buyingUnit, false)
        buyingUnit.addBonus(utype.getBonus())
        buyingUnit = null
       
    function showShop(unit buyer)
        if buyingUnit == null
            buyingUnit = buyer
            dialogBox.display(buyingUnit.getOwner(), true)
```

With the use of closures, the class instance and therefore `buyingUnit` are automatically available inside the lambda expression, which is not the case with code.
In that case the user would have to attach his data manually somehow. 
Using closures would also get rid of triggers completely.